tests/builders/test_design_spaces.py

```diff
@@ -191,3 +193,99 @@ def test_exceptions(basic_cartesian_space, simple_mixture_space):
             name='invalid join space 2',
             description=''
         )
+
+
+def test_oversize_warnings(large_joint_design_space):
+    """Test that warnings are raised"""
+    # Test oversized formulation grid
+    with pytest.raises(UserWarning,
+                       match="Product design grid contains 1562500000 grid points. "
```

Comment:
[Non-blocking] This match may be too specific.  I would just do `match="1562500000"` so that the text of the warning can be changed more easily.  Likewise for the other tests.

Reply:
Agreed - done in last push